### PR TITLE
meep: add new versions and additional variants incl. dependencies

### PR DESCRIPTION
**Pull request by muffgaga**

### CODE REVIEW

File: var/spack/repos/builtin/packages/meep/package.py [hunk 1]
```diff
@@ -106,6 +112,11 @@ def configure_args(self):
         if "+single" in spec:
             config_args.append("--enable-single")
 
+        if "+openmp" in spec:
+            config_args.append("--with-openmp")
+        else:
+            config_args.append("--without-openmp")
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/meep/package.py]: You can simplify this with one of the AutotoolsPackage helpers (see https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#with-or-without).

**Review by tldahlgren — COMMENTED** [review]
Confirmed all four sha256.

File: var/spack/repos/builtin/packages/meep/package.py (hunk 1 above)
Comment (muffgaga) [reply var/spack/repos/builtin/packages/meep/package.py]: Fixed && I also added a commit on top which simplifies two other trivial with/without options.

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**